docs/build-provenance-for-all-registries.md

```diff
@@ -0,0 +1,104 @@
+# Build Provenance for All Registries
+
+There’s a new security capability that registries can offer - build provenance - which verifiably links a package back to its source code and build instructions.
```

Comment:
I added "Since these links are publicly available, package consumers can verify the package contains what it says, instead of having to trust that the build process or source code wasn't compromised." to bring in elements of the SLSA model, while also incorporating @znewman01's feedback below.